### ISSUE TITLE
Support for nullish coalescing and optional chaining?

### ISSUE THREAD

**Issue by stefanuddenberg** [jshint/jshint#3443]
It would be nice for jshint to support those two features that are set to come to the language:

https://github.com/tc39/proposal-nullish-coalescing
https://github.com/tc39/proposal-optional-chaining

**Comment by jugglinmike** [comment]
You're right. These are independent features, so we should have separate tracking issues for them. Would you mind filing another issue for one of the two?

(Before implementing new language features, we require them to have reached "Stage 2" of TC-39's standardization process. Because both features requested here have reached "Stage 4", they both qualify. Here's a blog post explaining the policy: [A Formal Commitment to New Language Features](https://jshint.com/blog/new-lang-features/))

**Comment by stefanuddenberg** [comment]
Done!

**Comment by jugglinmike** [comment]
Thank you, Stefan :)

(For those following along, the new issue is gh-3448.)

**Comment by Magnogen** [comment]
Hi, not sure how far through this fix is to being pushed, but I get errors for nullish assignment as well (`??=`)

I end up getting lots of errors for one line, so using a lot of these ends up resulting in a `Too many errors. (98% scanned).` error :/

These are the errors that occur for each line of nullish assignment:
```
Expected an identifier and instead saw '='.
Unexpected '{a}'.
Expected an assignment or function call and instead saw an expression.
Missing semicolon.
Expected an assignment or function call and instead saw an expression. (yes this happens twice)
```

This is on the website, https://jshint.com/

**Comment by renebaudisch** [comment]
This is still open? Nice...
